src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -183,20 +191,53 @@ private bool VerifyCSharpPlugin(IProgressController controller, CancellationToke
 
-                this.host.ActiveSection?.UserNotifications?.ShowNotificationError(errorMessage, NotificationIds.BadServerPluginId, null);
-                notifications.ProgressChanged(errorMessage);
-                notifications.ProgressChanged(Strings.ConnectionResultFailure);
+            var projects = this.projectSystem.GetSolutionProjects().ToList();
+            var anyCSharpProject = projects.Any(MinimumSupportedServerPlugin.CSharp.ISupported);
+            var anyVbNetProject = projects.Any(MinimumSupportedServerPlugin.VbNet.ISupported);
 
-                AbortWorkflow(controller, cancellationToken);
-                return false;
+            string errorMessage;
+            if ((IsCSharpPluginSupported && anyCSharpProject) ||
+                (IsVBNetPluginSupported && anyVbNetProject))
+            {
+                return true;
+            }
+            else if (!IsCSharpPluginSupported && !IsVBNetPluginSupported)
+            {
+                errorMessage = Strings.ServerHasNoSupportedPluginVersion;
+            }
+            else if (projects.Count == 0)
+            {
+                errorMessage = Strings.SolutionContainsNoSupportedProject;
             }
+            else if (IsCSharpPluginSupported && !anyCSharpProject)
+            {
+                errorMessage = string.Format(Strings.OnlySupportedPluginHasNoProjectInSolution, Language.CSharp.Name);
+            }
+            else
+            {
+                errorMessage = string.Format(Strings.OnlySupportedPluginHasNoProjectInSolution, Language.VBNET.Name);
+            }
+
+            this.host.ActiveSection?.UserNotifications?.ShowNotificationError(errorMessage, NotificationIds.BadServerPluginId, null);
+            VsShellUtils.WriteToSonarLintOutputPane(this.host, Strings.SubTextPaddingFormat, errorMessage);
+            notifications.ProgressChanged(Strings.ConnectionResultFailure);
+
+            AbortWorkflow(controller, cancellationToken);
+            return false;
+        }
+
+        private bool VerifyPluginSupport(ServerPlugin[] plugins, MinimumSupportedServerPlugin minimumSupportedPlugin)
+        {
+            var plugin = plugins.FirstOrDefault(x => StringComparer.Ordinal.Equals(x.Key, minimumSupportedPlugin.Key));
+            var isPluginSupported = !string.IsNullOrWhiteSpace(plugin?.Version) && VersionHelper.Compare(plugin.Version, minimumSupportedPlugin.MinimumVersion) >= 0;
+
+            var pluginSupportMessageFormat = string.Format(Strings.SubTextPaddingFormat, isPluginSupported ? Strings.SupportedPluginFoundMessage : Strings.UnsupportedPluginFoundMessage);
+            VsShellUtils.WriteToSonarLintOutputPane(this.host, pluginSupportMessageFormat, minimumSupportedPlugin.ToString());
 
-            return true;
+            return isPluginSupported;
         }
 
         private static void AbortWorkflow(IProgressController controller, CancellationToken token)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this parameter "token", whose value is ignored in the method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1172)